elephant_vending_machine/views.py

```diff
@@ -48,17 +52,36 @@ def upload_image():
     will be returned
 
     Returns:
-        HTTP response 200 with payload 'File received' or
-        HTTP response 500 with payload 'No image file in request.'
-
+        HTTP response 200 with payload 'Success: Image saved.' or
+        HTTP response 400 with payload describing the issue with the request.
     """
 
     response = ""
-    if 'file' in request.files:
-        response = "File received."
+    response_code = 400
+    if 'file' not in request.files:
+        response = "Error with request: No file field in body of request."
     else:
-        response = "No image file in request."
-    return response
+        file = request.files['file']
+        if file.filename == '':
+            response = "Error with request: File field in body of response with no file present."
+        elif file and allowed_file(file.filename):
+            filename = secure_filename(file.filename)
+            file.save(os.path.join(APP.config['IMAGE_UPLOAD_FOLDER'], filename))
```

Comment:
This is my own python ignorance showing here, but what happens if the config value for file directory is set to a directory that doesn't exist? Will this create the necessary directory structure? If not, is that something we could easily add or would it be complicated?

Reply:
If the directory doesn't exist it throws an error, I can look into auto-creating the directory if it doesn't exist. However, if the folder we're submitting images to needs to be synced with the other pis, I don't think it would make much sense to be able to change it to an arbitrary directory, as then that directory would need to be set up to sync.

Reply:
Creating the directory if it doesn't exist is trivial

Reply:
I have a task this sprint to sync the images and while I was thinking of using Syncthing, it may be better for us to have this endpoint explicitly sync the images. That way there isn't any configuration that lives in other random programs even though that might save us some small amount of work. Since creating the directory is trivial, I think we should do that and I'll add other config variables for remote image directory location and add some code to handle the syncing.